File: src/main/java/com/gitlab/daring/image/command/feature/MSERCommand.java
package com.gitlab.daring.image.command.feature;

import com.gitlab.daring.image.command.BaseCommand;
import com.gitlab.daring.image.command.CommandEnv;
import com.gitlab.daring.image.command.parameter.IntParam;
import org.bytedeco.javacpp.opencv_features2d.MSER;

class MSERCommand extends BaseCommand {

    //TODO support all params
    final IntParam delta = intParam(5, "0-100");
    final IntParam minArea = intParam(60, "0-1000");
    final IntParam maxArea = intParam(14400, "0-100000");
    final MSER detector = MSER.create();

    MSERCommand(String... args) {
        super(args);
    }

    @Override
    public void execute(CommandEnv env) {
        detector.setDelta(delta.v);
        detector.setMinArea(minArea.v);
        detector.setMaxArea(maxArea.v);
        env.featureDetector = detector;
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/feature/MSERCommand.kt
package com.gitlab.daring.image.command.feature

import com.gitlab.daring.image.command.CommandEnv
import com.gitlab.daring.image.command.KBaseCommand
import org.bytedeco.javacpp.opencv_features2d.MSER

internal class MSERCommand(vararg args: String) : KBaseCommand(*args) {

    //TODO support all params
    val delta = intParam(5, "0-100")
    val minArea = intParam(60, "0-1000")
    val maxArea = intParam(14400, "0-100000")
    val detector = MSER.create()

    override fun execute(env: CommandEnv) {
        detector.delta = delta.v
        detector.minArea = minArea.v
        detector.maxArea = maxArea.v
        env.featureDetector = detector
    }

}
